database/plugin/metadata/sqlite/utxo.go

```diff
@@ -206,14 +206,16 @@ func (d *MetadataStoreSqlite) SetUtxo(
 	stake []byte, // stake
 	amount uint64, // amount
 	assets *lcommon.MultiAsset[lcommon.MultiAssetTypeOutput], // assets (multi-asset)
+	transactionId *uint,
 	txn *gorm.DB,
 ) error {
 	tmpUtxo := models.Utxo{
-		TxId:       txId,
-		OutputIdx:  idx,
-		AddedSlot:  slot,
-		PaymentKey: payment,
-		StakingKey: stake,
+		TxId:          txId,
+		OutputIdx:     idx,
+		AddedSlot:     slot,
+		PaymentKey:    payment,
+		StakingKey:    stake,
+		TransactionID: transactionId,
```

Comment:
Remove the TransactionID here. It's not the given TxId. It's a little confusing because of the overloading of terms. `TransactionID` in `Utxo` refers to the internal database ID field (primary key) on the Transaction table. It is not the "Transaction ID" as we refer to it in Cardano, which is the hash of the transaction body.

This field would normally be populated by GORM when we save it while embedded in another structure. I'll show you how that works in another pass.